webpack/components/extensions/Hosts/BulkActions/BulkChangeHostCollectionsModal/BulkChangeHostCollectionsModal.js

```diff
@@ -0,0 +1,323 @@
+
+    dispatch(addToast({ type: 'success', message }));
+    dispatch(APIActions.get({
+      key: API_REQUEST_KEY,
+      url: foremanUrl(HOSTS_API_PATH),
+    }));
+    handleModalClose();
+  };
+
+  const handleSave = () => {
+    const requestBody = {
+      included: {
+        search: fetchBulkParams(),
+      },
+      host_collection_ids: HCIds,
+      organization_id: orgId,
+    };
+
+    if (addRadioChecked) {
+      dispatch(bulkAddHostCollections(requestBody, handleSuccess, handleError));
+    } else {
+      dispatch(bulkRemoveHostCollections(requestBody, handleSuccess, handleError));
+    }
+  };
+
+  const isLoadingHostCollections = hostCollectionsLoadingStatus === STATUS.PENDING;
+  const isSaveDisabled = HCIds.length === 0 || isLoadingHostCollections;
+
+  const modalActions = [
+    <Button
+      key="save"
+      ouiaId="bulk-change-host-collections-modal-save-button"
+      variant="primary"
+      onClick={handleSave}
+      isDisabled={isSaveDisabled}
+      isLoading={isLoadingHostCollections}
+    >
+      {__('Save')}
+    </Button>,
+    <Button
+      key="cancel"
+      ouiaId="bulk-change-host-collections-modal-cancel-button"
+      variant="link"
+      onClick={handleModalClose}
+    >
+      {__('Cancel')}
+    </Button>,
+  ];
+
+  const columns = {
+    name: {
+      title: __('Host collection'),
+      wrapper: hc => hc?.name || '',
+      isSorted: true,
+      weight: 50,
+    },
+    limit: {
+      title: __('Limit'),
+      wrapper: (hc) => {
+        const totalLimit = hc?.unlimited_hosts ? 'unlimited' : (hc?.max_hosts || 0);
+        const count = hc?.total_hosts || 0;
+        return `${count}/${totalLimit}`;
+      },
+      isSorted: false,
+      weight: 100,
+    },
+    description: {
+      title: __('Description'),
+      wrapper: hc => hc?.description || '',
+      isSorted: false,
+      weight: 150,
+    },
+  };
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onClose={handleModalClose}
+      onEscapePress={handleModalClose}
+      title={__('Change host collections')}
+      width="60%"
+      position="top"
+      actions={modalActions}
+      id="bulk-update-host-collections-modal"
+      key="bulk-update-host-collections-modal"
+      ouiaId="bulk-update-host-collections-modal"
+    >
+      <TextContent>
+        <Text ouiaId="bulk-update-host-collections-text">
+          {__(`Select host collection(s) to change their associations with selected host(s).
+            Changing host collection will affect all your selected hosts.
+            Some hosts may already be in your chosen collection(s).`)}
+        </Text>
+      </TextContent>
+      <h3><strong>{__('Select action')}</strong></h3>
+      <div style={{ display: 'flex', alignItems: 'baseline' }}>
+        <Radio
+          isChecked={addRadioChecked}
+          onChange={(_e, checked) => setAddRadioChecked(checked)}
+          name="radio-add"
+          label={__('Add to host collection(s)')}
```

Comment:
```suggestion
          label={__('Add to host collections')}
```
I think it's just as readable without the indeterminate plural. Or you could get fancy and do a proper translated plural based on the number of selected host collections -- up to you.